### PR TITLE
Improve tests for ActiveRecord::ConnectionAdapters::ConnectionPool co…

### PR DESCRIPTION
**Pull request by tgxworld**
…nfigurations.

### CODE REVIEW

**Comment by tgxworld** [comment]
Failure is unrelated.

File: activerecord/test/cases/connection_pool_test.rb [hunk 1]
```diff
@@ -156,6 +156,45 @@ def test_reap_inactive
...
+        @pool.checkin(idle_conn)
+
+        def idle_conn.seconds_idle
+          0.01
+        end
+
+        @pool.flush
+        assert_equal 1, @pool.connections.length
+
+        def idle_conn.seconds_idle
```
[the remark this answers is not in the document]
Comment (tgxworld) [reply activerecord/test/cases/connection_pool_test.rb]: @kamipo Where were you seeing the warning? I'm not seeing it in my console but I refactored the test to another pattern which I think will prevent the warning.

File: activerecord/test/cases/connection_pool_test.rb [hunk 2]
```diff
@@ -91,7 +91,9 @@ def test_active_connection_in_use
       end
 
       def test_full_pool_exception
+        @pool.checkout_timeout = 0.001 # no need to delay test suite by waiting the whole full default timeout
```
Comment (tgxworld) [inline activerecord/test/cases/connection_pool_test.rb]: This test was waiting the full 5 seconds.

**Comment by tgxworld** [comment]
@kamipo Updated :) Build failure is unrelated again.

screenshot from 2018-08-20 08-43-41

**Comment by kamipo** [comment]
Can you address a Code Climate issue?

image

File: activerecord/test/cases/connection_pool_test.rb [hunk 3]
```diff
@@ -166,9 +210,10 @@ def test_flush
 
         assert_equal 3, @pool.connections.length
 
-        def idle_conn.seconds_idle
```
Comment (tgxworld) [inline activerecord/test/cases/connection_pool_test.rb]: This was causing code climate checks to fail.

**Comment by tgxworld** [comment]
@kamipo  Updated :)

**Comment by tgxworld** [comment]
Thank you for reviewing :)

**Pull request merged**